src/classes/BGE_Template_CTRL.cls

```diff
@@ -0,0 +1,199 @@
+            String namespacedObject = UTIL_Namespace.StrTokenRemoveNSPrefix(dfr.name);
+            if (!bannedFields.contains(namespacedObject) && dfr.inlineHelpText != null) {
+                Field fld = new Field();
+                if (namespacedObject == 'Donation_Donor__c') {
+                    fld.sObjectName = 'Opportunity';
+                } else {
+                    fld.sObjectName = dfr.inlineHelpText.split('\\.')[0];
+                }
+                fld.name = (dfr.getLabel().contains(' ') && dfr.getLabel().split(' ',2)[0] == fld.sObjectName) ? dfr.getLabel().split(' ',2)[1] : dfr.getLabel() ;
+                fields.add(fld);
+            }
+        }
+        return fields;
+    }
+    /*******************************************************************************************************
+    * @description Retrieves the object and field labels for the component.
+    * @return Map of Batch Template object and field name keys to their labels.
+    */
+    @AuraEnabled
+    public static Map<String, String> getBatchTemplateLabels() {
+
+        Map<String, String> labels = new Map<String, String> {
+            'batchTemplateLabel' => Schema.SObjectType.Batch_Template__c.getLabel(),
+            'description' => Schema.Batch_Template__c.Description__c.getDescribe().getLabel(),
+            'totalPromptHelp' => Schema.Batch_Template__c.Display_Total_Prompt__c.getDescribe().getInlineHelpText(),
+            'totalPromptLabel' => Schema.Batch_Template__c.Display_Total_Prompt__c.getDescribe().getLabel(),
+            'totalMatchHelp' => Schema.Batch_Template__c.Require_Total_Match__c.getDescribe().getInlineHelpText(),
+            'totalMatchLabel' => Schema.Batch_Template__c.Require_Total_Match__c.getDescribe().getLabel()
+        };
+
+        return labels;
+    }
+
+    /*******************************************************************************************************
+    * @description Saves the Template Details information with the structured list of active fields.
+    * @return void.
+    */
+    @AuraEnabled
+    public static void saveTemplate(String templateInfo, String activeFields) {
+        try {
+            TemplateDetailsModel templateDetails = (TemplateDetailsModel) JSON.deserialize(templateInfo, TemplateDetailsModel.class);
+
+            Batch_Template__c batchTemplate = new Batch_Template__c();
+            batchTemplate.Name = templateDetails.name;
+            batchTemplate.Description__c = templateDetails.description;
+            batchTemplate.Display_Total_Prompt__c = templateDetails.displayTotalPrompt;
+            batchTemplate.Require_Total_Match__c = templateDetails.requireTotalMatch;
+            batchTemplate.Active_Fields__c = activeFields;
+
+            insert batchTemplate;
+        } catch (Exception ex) {
+            //TODO: is BDI right to use here?
+            ERR_Handler.processError(ex, ERR_Handler_API.Context.BDI);
```

Comment:
We might want our own BGE context, but for now this is probably fine. @force2b when do we add a new ERR_Handler context?